substrate/frame/support/src/dispatch.rs

```diff
@@ -687,6 +687,30 @@ impl<T> PaysFee<T> for (u64, Pays) {
 	}
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct VersionedCall<Call> {
+    pub call: Call,
```

Comment:
I think better to make it private and only retrievable through validate method and bypass_validation